frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/network-attachment-definition.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { sortable } from '@patternfly/react-table';
+import { ListPage, Table, TableData, TableRow } from '@console/internal/components/factory';
+import { Kebab, ResourceKebab, ResourceLink } from '@console/internal/components/utils';
+import { NamespaceModel } from '@console/internal/models';
+import { K8sResourceKindReference } from '@console/internal/module/k8s';
+import { getName, getNamespace, getUID } from '@console/shared/src';
+import { getConfig } from '../../utils';
+import { NetworkAttachmentDefinitionModel } from '../../models';
+import { NetworkAttachmentDefinitionKind } from '../../types';
+
+const { common } = Kebab.factory;
+const menuActions = [...common];
+
+export const NetworkAttachmentDefinitionReference: K8sResourceKindReference =
+  'NetworkAttachmentDefinition';
+
+const tableColumnClasses = [
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  classNames('col-lg-4', 'col-md-4', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  Kebab.columnClass,
+];
+
+const NetworkAttachmentDefinitionsHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Namespace',
+      sortField: 'metadata.namespace',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
```

Comment:
the type sorting doesn't seem to work - tested with  `all projects`

Reply:
I tested it with `all projects` as well. It seemed to work fine for me. Did it not change at all for you?

Reply:
It switches between these two states, neither of which is sorted by type:

![a](https://user-images.githubusercontent.com/3648838/66116619-4f220300-e5d3-11e9-9b39-a3dbeb9bf2e4.png)

![b](https://user-images.githubusercontent.com/3648838/66116626-52b58a00-e5d3-11e9-9710-9fa530434338.png)

Reply:
This should be fixed now.